// Copyright (c) 2020 The Orbit Authors. All rights reserved.
// Use of this source code is governed by a BSD-style license that can be
// found in the LICENSE file.

syntax = "proto3";

package orbit_grpc_protos;

// Hello world test code
service CaptureClientGgpService {
  rpc StartCapture (StartCaptureRequest) returns (StartCaptureResponse) {}
  rpc StopAndSaveCapture (StopAndSaveCaptureRequest) returns (StopAndSaveCaptureResponse) {}
  rpc UpdateSelectedFunctions (UpdateSelectedFunctionsRequest) returns (UpdateSelectedFunctionsResponse) {}
  rpc ShutdownService (ShutdownServiceRequest) returns (ShutdownServiceResponse) {}
}

message StartCaptureRequest {}

message StartCaptureResponse {}

message StopAndSaveCaptureRequest {}

message StopAndSaveCaptureResponse {}

message UpdateSelectedFunctionsRequest {
  repeated string functions = 1;
}

message UpdateSelectedFunctionsResponse {}

message ShutdownServiceRequest {}

message ShutdownServiceResponse {}
